src/NHibernate/Proxy/AbstractProxyFactory.cs

```diff
@@ -42,8 +43,28 @@ protected bool IsClassProxy
 			SetIdentifierMethod = setIdentifierMethod;
 			ComponentIdType = componentIdType;
 			OverridesEquals = ReflectHelper.OverridesEquals(persistentClass);
+			IsClassProxy = isClassProxy;
```

Comment:
To avoid breaking external implementations of `AbstractProxyFactory`, we should call the obsolete method here, which would do most of the initialization. Then the new method would override `IsClassProxy` value with the new parameter value.

Reply:
Done